### ISSUE TITLE
Non-reproducible minified assets files (js/css)

### ISSUE THREAD

**Issue by Kdecherf** [wallabag/wallabag#2651]
During my work on #2622 I noticed that `style.min.css` was different between the top of 2.2 branch and my local copy of the 2.2 branch.

The minified file depends on the version of node packages used to perform assets compilation. Sadly it appears that the minified file differs even between minor versions of a certain package. This can lead to inconsistent changes to be committed by contributors.

It could be interesting to avoid the commit of minified files in the development workflow and use a workflow similar to what is done for Composer: we could commit minified files only once during release.
This new workflow also could ease rebase tasks during PR or merge between release branches as long as the main conflict I observed concerns the minified files themselves.

On the implementation side there are several possible options:
- list all assets in templates in dev mode instead of minified files
- perform a concat only in dev mode (_ie, remove the minify from grunt task tests or create a new task for dev only_) and save the output as `style.min.css` (_or whatever for js files too_). This file should be added to `.gitignore`
- freeze npm dependencies with `shrinkwrap` or something similar

Imo the second option is the easier to do.
RFC

**Comment by tcitworld** [comment]
Second option suits me too. Are you wanting to submit a PR ?

P.S : I'll have to check some js files aren't already concated.

**Comment by Kdecherf** [comment]
> Are you wanting to submit a PR ?

I'm not sure I will have time to handle this so I let others taking care of it :)

> How would users get it in that case since we distribute through git ?

Like for `composer.lock`, normal users will have the minified file provided in the release archive and developers (_or users from git_) will have to launch a grunt task to make a quick concat file.

**Comment by nicosomb** [comment]
Second option too :+1:

**Comment by tcitworld** [comment]
> and developers (or users from git) will have to launch a grunt task to make a quick concat file.

Nope, we tried that once, that's a very bad idea. And most of users didn't even had node on their systems.

**Comment by Kdecherf** [comment]
> Nope, we tried that once, that's a very bad idea. And most of users didn't even had node on their systems.

We could document the way to retrieve the last minified files from release commit or release archive (_and/or add a script to automatically fetch it_).

**Comment by tcitworld** [comment]
I find that more complicated. My idea is to have in twig a `%devcsspath%` which is used if we are in a dev environment.

**Comment by Kdecherf** [comment]
It could be an idea but contributors should not commit changes to `style.min.css` anyway.

**Comment by j0k3r** [comment]
Does the webpack jump fix that kind of problem?

**Comment by tcitworld** [comment]
We have the dev builds so that should be ok. Prod build should only be generated if something's changed in the assets.
